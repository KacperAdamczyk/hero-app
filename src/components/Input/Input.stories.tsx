import React from 'react';
import { text } from '@storybook/addon-knobs';
import { Formik, Form } from 'formik';

import { Input } from './Input';
import { action } from '@storybook/addon-actions';

export default {
  title: 'Input',
  component: Input,
};

export const Default = () => {
  const label = text('label', 'Field');

  return (
    <Formik
      onSubmit={action('onSubmit')}
      initialValues={{
        field: '',
      }}
    >
      <Form>
        <Input name="field" label={label} />
      </Form>
    </Formik>
  );
};
