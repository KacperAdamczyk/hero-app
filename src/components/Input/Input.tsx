import React, { FC } from 'react';
import { useField, FieldConfig, ErrorMessage } from 'formik';
import styled from '@emotion/styled';

import { colors } from 'styling';

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Label = styled.label`
  color: ${colors.textLight};
  margin-bottom: 5px;
  font-size: 0.9em;
  font-weight: bold;
`;

export const NativeInput = styled.input`
  height: 42px;
  padding: 0 15px;
  background-color: white;
  border: none;
  outline: none;
  border-radius: 5px;
  font-size: 1em;
`;

export const FieldError = styled.div`
  color: red;
  opacity: 0.8;
  font-size: 0.8em;
  margin-top: 3px;
`;

interface Props {
  label: string;
}

export const Input: FC<FieldConfig<string> & Props> = ({ label, ...props }) => {
  const { name } = props;

  const [field] = useField(props);

  return (
    <Container>
      <Label htmlFor={name}>{label}</Label>
      <NativeInput id={name} {...field} />
      <ErrorMessage name={name} component={FieldError} />
    </Container>
  );
};
