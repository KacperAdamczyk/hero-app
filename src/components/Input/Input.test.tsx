import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import { Formik } from 'formik';

import { Input } from './Input';

describe('Input', () => {
  test('should display a value', () => {
    const { getByDisplayValue, container } = render(
      <Formik initialValues={{ input: 'foo' }} onSubmit={() => {}}>
        <Input name="input" label="Input" />
      </Formik>,
    );

    const input = getByDisplayValue('foo');

    expect(input).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should fire have provided value', () => {
    const { getByLabelText, getByDisplayValue } = render(
      <Formik initialValues={{ input: '' }} onSubmit={() => {}}>
        <Input name="input" label="Input" />
      </Formik>,
    );

    const input = getByLabelText('Input');
    act(() => {
      fireEvent.input(input, { target: { value: 'bar' } });
    });

    expect(getByDisplayValue('bar')).toBeInTheDocument();
  });
});
