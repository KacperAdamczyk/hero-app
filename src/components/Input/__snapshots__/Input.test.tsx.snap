// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should display a value 1`] = `
<div
  class="css-1iyoj2o"
>
  <label
    class="css-1o8lxzy"
    for="input"
  >
    Input
  </label>
  <input
    class="css-19e9lyz"
    id="input"
    name="input"
    value="foo"
  />
</div>
`;
